core/test/scripts/run-mocha-tests.js

```diff
@@ -277,33 +272,66 @@ function exit({numberFailures, numberMochaInvocations}) {
  * @param {number} invocationNumber
  */
 async function runMocha(tests, mochaArgs, invocationNumber) {
-  process.env.LH_FAILED_TESTS_FILE = `${failedTestsDir}/output-${invocationNumber}.json`;
+  const failedTestsFile = `${failedTestsDir}/output-${invocationNumber}.json`;
+  const notRunnableTestsFile = `${failedTestsDir}/output-${invocationNumber}-nr.json`;
+  process.env.LH_FAILED_TESTS_FILE = failedTestsFile;
 
   const rootHooksPath = mochaArgs.require || '../test-env/mocha-setup.js';
   const {rootHooks} = await import(rootHooksPath);
 
-  try {
-    const mocha = new Mocha({
-      rootHooks,
-      timeout: 20_000,
-      bail: mochaArgs.bail,
-      grep: mochaArgs.grep,
-      forbidOnly: mochaArgs.forbidOnly,
-      // TODO: not working
-      // parallel: tests.length > 1 && mochaArgs.parallel,
-      parallel: false,
-      retries: mochaArgs.retries,
-    });
-
-    // @ts-expect-error - not in types.
-    mocha.lazyLoadFiles(true);
-    for (const test of tests) mocha.addFile(test);
-    await mocha.loadFilesAsync();
-    return await new Promise(resolve => mocha.run(resolve));
-  } catch (err) {
-    console.error(err);
-    return 1;
+  const mocha = new Mocha({
+    rootHooks,
+    timeout: 20_000,
+    bail: mochaArgs.bail,
+    grep: mochaArgs.grep,
+    forbidOnly: mochaArgs.forbidOnly,
+    // TODO: not working
+    // parallel: parsableTests.length > 1 && mochaArgs.parallel,
+    parallel: false,
+    retries: mochaArgs.retries,
+  });
+
+  const notRunnableTests = [];
+  const parsableTests = [];
+  for (const test of tests) {
+    try {
+      mocha.files = [test];
+      await mocha.loadFilesAsync();
+      parsableTests.push(test);
+    } catch (e) {
+      notRunnableTests.push({
+        file: path.relative(LH_ROOT, test),
+        title: '',
+        error: `Failed to parse module: ${e}`,
+      });
+    }
+  }
+  mocha.files = [];
+
+  let failingTests = 0;
```

Comment:
nit: `numFailingTest` or `failingTestCount` to differentiate from `notRunnableTests` which is an array